frontend/include/chpl/resolution/resolution-types.h

```diff
@@ -2273,16 +2855,31 @@ class ResolvedFunction {
   // the return type computed for this function
   types::QualifiedType returnType_;
 
+  // If a call to a child function is made within this function, then the
+  // resulting 'ResolvedFunction' for the child is stored within this map.
+  // The primary key is the POI trace, and the secondary key is signature
+  // and initial PoiInfo. This mirrors the way functions are cached.
+  PoiTraceToChildMap poiTraceToChild_;
```

Comment:
So, a nested function could be instantiated differently than the outer function, so need its own POI info? Why use a PoiInfo::Trace rather than storing a new PoiInfo? Is it because of some query-based discipline with the PoiInfo that needs to be avoided here?

But SigAndInfoToChildPtrMap uses PoiInfo? I am not catching on to what "initial PoiInfo" means. Can you remind me?

Reply:
So all I was trying to do here was mimic how caching of generics is implemented using queries. The map is just replacing the role of the query cache in `resolveFunctionByPois`. If you think I can simplify what's going on here I'm all ears.

The initial PoiInfo is the PoiInfo that is input to `resolveFunctionByInfoQuery`.